src/context/LoadingProvider.jsx

```diff
@@ -0,0 +1,17 @@
+import { createContext, useState} from 'react';
+
+
+export const LoadingContext = createContext();
+
+
+export const LoadingProvider = ({ children }) => {
+  const [loading, setLoading] = useState(false);
+
+  return (
+    <LoadingContext.Provider value={{ loading, setLoading }}>
+      {children}
+    </LoadingContext.Provider>
+  );
+};
+
+
```

Comment:
This is not an ideal use of context. If a single part of the app needs to load something, then this causes your whole app to be in a loading state which is not ideal. Loading states should be local to the component that is fetching data and need to show something loading.